wallets/provider-trezor/src/index.ts

```diff
@@ -29,18 +34,44 @@ export const init = (environments: Environments) => {
 export const getInstance = getTrezorInstance;
 
 let isTrezorInitialized = false;
-export const connect: Connect = async () => {
+export const connect: Connect = async ({ namespaces }) => {
+  const results: ProviderConnectResult[] = [];
+
   const TrezorConnect = await getTrezorModule();
 
-  if (!isTrezorInitialized) {
-    await TrezorConnect.init({
-      lazyLoad: true, // this param will prevent iframe injection until TrezorConnect.method will be called
-      manifest: trezorManifest,
-    });
+  const evmNamespace = namespaces?.find(
+    (namespaceItem) => namespaceItem.namespace === Namespace.Evm
+  );
+
+  if (evmNamespace) {
+    if (evmNamespace.derivationPath) {
+      setDerivationPath(
+        getTrezorNormalizedDerivationPath(evmNamespace.derivationPath)
+      );
 
-    isTrezorInitialized = true;
+      if (!isTrezorInitialized) {
+        await TrezorConnect.init({
+          lazyLoad: true, // this param will prevent iframe injection until TrezorConnect.method will be called
+          manifest: trezorManifest,
+        });
+
+        isTrezorInitialized = true;
+      }
+
+      const accounts = await getEthereumAccounts(getDerivationPath());
```

Comment:
Handle getDerivationPath() in the function itself. Its structure should be better same as Ledger, no need to offer it as a param.

Reply:
Tnx Fixed